### ISSUE TITLE
DynamicPayment formula is incorrect

### ISSUE THREAD

**Issue by MilesBHuff** [Zrips/Jobs#477]
# Description of issue

**tl;dr:  The default DynamicPayment formula should be `offset = ((population / jobs) / workers) - 1` instead of `offset = ((population / jobs) - workers) / 10`.**

The default DynamicPayment formula breaks down at large volumes, and worse is based on the wrong phenomenon.  

Its goal is to adjust the equilibrium price per available supply and demand, with each player on the server providing `1 / jobs` demand, and each worker in each job providing `1` supply for that job.  The default formula is `offset = ((population / jobs) - workers) / 10`, with `workers` being only for the current job, and this formula being run for each individual job on the server.

The problem is that instead of calculating the offset to the equilibrium price based on the ratio between supply and demand as it should be doing, it's calculating the offset by taking the difference between the supply and the demand and dividing it by an arbitrary 10.  While this *seems* at first glance to work with small population sizes, as soon as you scale up to anything large, the system completely breaks down.  Under the default system, when there is a supply of `1` item and demand for `10`, the offset is `90%`;  and when there is a supply of `10` items and demand for `100`, the offset is `900%`.  In each case, the supply:demand ratio is identical, and the resulting offset ought to be as well.  This S/D deficit-based system also produces outright nonsensical values:  if there are 200 players and 50 people in a job, the job earns *-333%* relative to normal.  That's *insane* and completely nonsensical.  There's a "fix" for this in the config with the `min`/`max` settings, but that's a band-aid solution for a fundamentally broken algorithm.

The goal is to adjust the equilibrium price per the supply and demand, not per the deficit between them.

If, rather than looking at the S/D deficit, we instead look at the ratio between supply and demand, things start to make a lot more sense.  We'll start by thinking about how much prices would intuitively/ideally change if the supply dropped and the demand stayed the same.  We'll have the price be at `100%` (a `0%` offset) when the demand is `1` and the supply is also `1`.  If that supply goes up to `2`, we have now doubled the supply without adjusting the demand.  Since there are now twice as many products available as before, the value of those products would assumably go down by half.  So at a demand of `1` and a supply of `2`, the price sits at `50%` (an offset of `-50%`).  This makes intuitive sense.  The formula for what we just did, is `offset = ((population / jobs) / workers) - 1`.  How does this new formula fare with the barrels example from earlier?  As desired, this new formula gets exactly the same offset for both the `1`:`10`::S:D scenario and the `10`:`100`::S:D scenario:  `900%` (This is actually a correct and reasonable value, as there is only `1/10` the normal supply -- a scenario that would drive prices up considerably.).

This new formula not only matches intuition, but it also avoids arbitrary parameters, like the `/ 10` of the original formula.  Even better:  this new formula has natural asymptotes at `-100%` and `infinity`, which means that the `min`/`max` band-aid is no longer necessary (although I suppose there's no harm in leaving it in).  I, too, was taken aback at first when I realized the formula could collapse into a singularity (*ie*, divide by zero).  However, this is only possible when the supply is `0` (*ie*, there are no players in that job.).  When the supply is `0`, then it means that the product you are trying to buy literally does not exist.  A product that does not exist has an infinite/undefined price, because it cannot be bought.  So, in actuality, the fact that this new formula can divide by zero is a feature, not a bug.

In summary, the current default DynamicPayment formula is fundamentally broken.  It requires arbitary math, it breaks down at large volumes, and it isn't even based on the right phenomenon.
We should adopt the correct formula for what DynamicPayment wants to do, which is adjust labor prices based on the ratios of supply and demand.  The formula I have proposed here does exactly that, and only that.

`offset = ((population / jobs) / workers) - 1`

# Relevant config

```
DynamicPayment:
    # Do you want to use dynamic payment dependent on how many players already working for jobs?
    # This can help automatically lift up payments for not so popular jobs and lower for most popular ones
    use: false
    equation: ((totalworkers / totaljobs) - jobstotalplayers)/10.0
    MaxPenalty: 25.0
    MaxBonus: 100.0
```

The above should be the below.

```
DynamicPayment:
    # Do you want to dynamically adjust pay according to supply and demand?
    # This will increase pay for unpopular jobs and decrease it for popular jobs,
    # in order to balance the two and prevent a single meta from winning out.
    # totalworkers:  The total number of players on the server who have jobs
    # totaljobs:  The number of jobs that are enabled
    # jobstotalplayers:  The number of people in each job
    use: false
    equation: totalworkers / totaljobs / jobstotalplayers - 1
    MaxPenalty: 100
    MaxBonus: 999
```

# Server info

**Type:**  Spigot

**Version:**  1.14.1

**Comment by Zrips** [comment]
Well original equation was made just to be there, as you can always change it to your own liking. Its not the best one, but it does adjust payments depending on how many workers you have in particular job. That one is linear, which is not the best one, yours is some what better one as it does change bonus/penalty in more gradual way when its reaching 0% which means that you can have less of fluctuation when you have balanced work force. Which is average worker count for each job. 
having too big bonus ir too big penalty is not good ether. Tho your equation cant really get any lower then 90% penalty. So yea, we can change default equation to yours and maybe add comment with old one, in case some one needs it.

**Comment by Zrips** [comment]
c188ec86581e189a43c6e2a7c46473d76192e89c
